src/firefly/js/visualize/PlotViewUtil.js

```diff
@@ -136,42 +136,33 @@ export function hasWCSProjection(ref) {
  * Return an array of plotId's that are in the plot group associated with the the pvOrId parameter.
  * @param {VisRoot} visRoot - root of the visualization object in store
  * @param pvOrId this parameter will take the plotId string or a plotView object
- * @param onlyIfPositionLocked
- * @param hasPlots
  * @returns {Array.<String>}
  */
-export function getPlotViewIdList(visRoot, pvOrId, onlyIfPositionLocked=true, hasPlots=false) {
+export function getPlotViewIdListByPositionLock(visRoot, pvOrId) {
     if (!pvOrId) return [];
-    const pv= (typeof pvOrId ==='string') ? getPlotViewById(visRoot,pvOrId) : pvOrId;
-    const locked= visRoot.positionLock;
-    if (!locked && onlyIfPositionLocked) return [pv.plotId];
-    const idList=  visRoot.plotViewAry.map( (pv) => pv.plotId);
-    if (!hasPlots) return idList;
-
-    return idList.filter( (id) => get(getPlotViewById(visRoot,id),'plots.length') );
+    const pv= isString(pvOrId) ? getPlotViewById(visRoot,pvOrId) : pvOrId;
+    if (!visRoot.positionLock) return [pv.plotId];
+    return visRoot.plotViewAry
+        .map( (pv) => pv.plotId)
+        .filter( (id) => get(getPlotViewById(visRoot,id),'plots.length') );
 }
-
 /**
  * Return an array of plotId's that are in the plot group associated with the the pvOrId parameter.
  * @param {VisRoot} visRoot - root of the visualization object in store
- * @param pvOrId this parameter will take the plotId string or a plotView object
- * @param onlyIfGroupLocked
- * @param hasPlots
+ * @param pvOrId this parameter will take the plotId string or a plotView objectgs
  * @returns {Array.<String>}
  */
-export function getPlotViewIdListInOverlayGroup(visRoot,pvOrId,onlyIfGroupLocked=true, hasPlots=false) {
+export function getPlotViewIdListInOverlayGroup(visRoot,pvOrId) {
     if (!pvOrId) return [];
     const pv= (typeof pvOrId ==='string') ? getPlotViewById(visRoot,pvOrId) : pvOrId;
     const gid= pv.plotGroupId;
     const group= getPlotGroupById(visRoot,gid);
-    const locked= hasOverlayColorLock(pv,group);
-    if (!locked && onlyIfGroupLocked) return [pv.plotId];
+    if (!hasOverlayColorLock(pv,group) ) return [pv.plotId];
     const idList=  visRoot.plotViewAry.filter( (pv) => pv.plotGroupId===gid).map( (pv) => pv.plotId);
-    if (!hasPlots) return idList;
-
     return idList.filter( (id) => get(getPlotViewById(visRoot,id),'plots.length') );
 }
```

Comment:
over overlay/color locking I don't think you need to filter out plotViews without plots. I think it is safer not to since that is basicly a `hasPlot=true` and it was never called that way for you use case.